src/ff/prime_field.rs

```diff
@@ -85,6 +85,52 @@ macro_rules! field_impl {
             }
         }
 
+        impl std::ops::BitAnd for $field {
+            type Output = Self;
+            fn bitand(self, rhs: Self) -> Self::Output {
+                Self(self.0 & rhs.0)
+            }
+        }
+
+        impl std::ops::BitAndAssign for $field {
+            fn bitand_assign(&mut self, rhs: Self) {
+                *self = *self & rhs;
+            }
+        }
+
+        impl std::ops::BitOr for $field {
```

Comment:
Is this right?

Consider `Fp31`.  What do you get if you invoke `Fp31(30) | Fp31(1)`?  The result based on the code below is `Fp31(31)`, which isn't valid.

You might think that implementing this in terms of `Fp31::from()` instead might fix the problem.  That would ensure that the answer is at least in the field (`Fp31(0)`), but I don't know if that is logically consistent with the definition of the type.

I don't see a way to make bit-level operations work for fields.  But perhaps we don't need to.

That is, while we might define arithmetic operations that operate on arrays of `Fp2`, there is no need to define bitwise operations on generic fields.